hw7: functions/functional.py

```diff
@@ -0,0 +1,58 @@
+################################################################################
+# this homework explores the functionality of functional programming in Python #
+################################################################################
+
+def sequential_map(*args):
+    result = args[-1]
+    for i in args[:-1]:
+        result = list(map(i, result))
+    return result
```

Comment:
\+ ещё можно воспользоваться распаковкой, это супер-удобно

`i` - ужасное имя для функции
```suggestion
    *functions, container = args
    for func in functions:
        result = map(i, result)
    return list(result)
```